xapian-core/common/serialise-double.cc

```diff
@@ -104,31 +103,38 @@ string serialise_double(double v){
 
     int exp;
     v = frexp(v, &exp);
-    v *= 2.0;
 
     if (exp == 0 && v == 0.0) {
-	result = 0;
+	result = 0.0;
```

Comment:
Why this change?  `result` is a `uint64_t` so `0` is a clearer zero value for it than `0.0`.

Reply:
Sorry for the unwanted change. In the latest commit, I removed this condition altogether because I was accidentally performing this check twice. I now kept it to check at the start if V equals 0.0, and return 0 if it does.

Reply:
Oh yes, this can only happen when v is 0.  Well spotted.